clang/test/Misc/target-invalid-cpu-note/aarch64.c

```diff
@@ -0,0 +1,90 @@
+// Use --implicit-check-not to ensure no additional CPUs are in this list
+
+// RUN: not %clang_cc1 -triple arm64--- -target-cpu not-a-cpu -fsyntax-only %s 2>&1 | FileCheck %s --implicit-check-not={{[a-zA-Z0-9]}}
+// RUN: not %clang_cc1 -triple arm64--- -tune-cpu not-a-cpu -fsyntax-only %s 2>&1 | FileCheck %s --implicit-check-not={{[a-zA-Z0-9]}}
+
+// CHECK: error: unknown target CPU 'not-a-cpu'
+// CHECK-NEXT: note: valid target CPU values are:
+// CHECK-SAME: a64fx,
```

Comment:
clang/test/Driver/linux-cross.cpp uses the `ARCH_I686-SAME: {{^}} xxx` pattern to check the content on the same line. You can use the same pattern. I think it's useful to pack some elements on the same line, e.g. `{{^}}, ampere1, ampere1a, ampere1b`

Reply:
I can use the caret pattern, but i'm keen to leave each CPU on one line to make diffs easier and clearer (so there's no rewrapping diffs)

Reply:
Tbh, the `--implicit-check-not` method looks cleaner to me than having `{{^}}` on every line. But it only works because there is so little output for this test to it can all be checked. There might be a case for adding (not in this PR) something like `CHECK-SAME-NEXT:` which would be useful across a large class of tests. The `-target-feature` tests could benefit from this too.

Reply:
Instead of having every CPU one the same line, you can group them by families like:
```
{{^}}, ampere1, ampere1a, ampere1b
{{^}}, fooX, fooY, fooZ
```

which still looks quite good.

Word diff tools are quite robust, so I think strictly avoiding packing is not necessary.

Reply:
What would be the advantage though? Fewer test lines? It's more work to group them. And sometimes we have downstream additions to processor families which would make resolving diffs more difficult if they appear mid-group.

Reply:
I am not going to group them, I think one per line is more maintainable long-term for a few different reasons - it means no rewrapping diffs, no having to reverse engineer how someone grouped things, and that it's very obvious what to do for new CPUs.

I have taken the advice on using `{{^}}`